crypto/provider_core.c

```diff
@@ -28,6 +28,89 @@
 # include <openssl/self_test.h>
 #endif
 
+/*
+ * This file defines and uses a number of different structures:
```

Comment:
This is just a great write up. Great work, Matt!

Reply:
Thanks!